### PR TITLE
LVUIDF-28: added updated version of Event entity.

### PR DESCRIPTION
**Pull request by irynar**
- [x] Add model, controller, migration, react router for Events.
- [x] Add react components for Events CRUD.

To review:
- [x] @rtriska 
- [ ] @vyshnevska 

Optional:
- [ ] @yurko 

To merge:
- [ ] @izagor
- [x] @forecome

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,4 +1,10 @@
 Rails.application.routes.draw do
   root 'application#index'
   get '/*path', to: 'application#index'
```
Comment (yurko) [inline config/routes.rb]: get '/*path', to: 'application#index' should be the last one to make API works properly

File: app/assets/javascripts/components/events/Event.js.jsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        <FieldView label="Date" value={this.state.event.date} formatter={dateFormatter} />
+        <FieldView label="Capacity" value={this.state.event.capacity} unit="people" />
+        <FieldView label="Place" value={this.state.event.place} />
+        <FieldView label="Location" value={[this.state.event.latitude, this.state.event.longitude]} formatter={mapFormatter} />
+      </div>
+      );
+  }
+});
+
+function dateFormatter(date) {
```
Comment (rtriska) [inline app/assets/javascripts/components/events/Event.js.jsx]: Extract those two functions below into helpers. Create directory with helpers, create js file with proper name and move there whose functions. Also put them into name spase 'helpers' -> 'FormatHelpers'
Just add it here https://github.com/dark-side/language-school/blob/develop/app/assets/javascripts/app.js.jsx :
```javascript
var Bb = {
  Models: {},
  Collections: {},
  Helpers: {}
};
```

**Comment by irynar** [comment]
I added FormatHelpers and some temporary changes for displaying events. Please review and merge.

File: app/views/application/index.html.erb [hunk 3]
```diff
@@ -0,0 +1 @@
+<%= react_router 'MyRoutes', 'HashLocation' %>
```
Comment (yurko) [inline app/views/application/index.html.erb]: we agreed to use no extra view for placing ```react_router``` helper. We replaced ```yield``` with that helper in ```layouts/application.html.haml```. See  https://github.com/dark-side/language-school/blob/develop/app/views/layouts/application.html.haml. So uncomment ```render text: '', layout: true``` in `ApplicationController` to render only layout

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -1,6 +1,7 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
   def index
-    render text: '', layout: true
+    @location_path = "/#{params[:path]}"
+    # render text: '', layout: true
```
Comment (yurko) [inline app/controllers/application_controller.rb]: is needed to be uncommented

File: app/controllers/application_controller.rb [hunk 5]
```diff
@@ -1,6 +1,7 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
   def index
-    render text: '', layout: true
+    @location_path = "/#{params[:path]}"
```
Comment (yurko) [inline app/controllers/application_controller.rb]: variable ```@location_path``` is used nowhere. It is needed only when prerendering React components at server. See https://github.com/mariopeixoto/react-router-rails/blob/master/examples/react-router-example/app/views/application/index.html.erb

File: app/assets/javascripts/components/events/Event.js.jsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
+var Event = React.createClass({
+  getInitialState: function () {
+    return {
+      eventId: 0,
```
Comment (rtriska) [inline app/assets/javascripts/components/events/Event.js.jsx]: why do we store 'evetId' and 'event' in different variables? If it's the same then we easely can access eventId on event. Lik event.id... Does it make sence?

File: app/assets/javascripts/components/events/Events.js.jsx [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    }, this);
+    this._events.fetch();
+  },
+  componentWillUnmount: function () {
+    this._events.off('update');
+  },
+  render: function () {
+    var eventsList = this.state.events.map(function (event, index) {
+      return (<Link to={'/events/' + event.id} className="list-group-item" key={index}>{event.name}</Link>);
+//      <li><Link to={`/users/${users.id}`}>{user.name}</Link></li>
```
Comment (rtriska) [inline app/assets/javascripts/components/events/Events.js.jsx]: if we have line which we don't need we just delete it :)

File: app/assets/javascripts/components/events/Event.js.jsx (hunk 6 above)
Comment (irynar) [inline app/assets/javascripts/components/events/Event.js.jsx]: eventId I get from React Router. Then this eventId is used for fetching concrete event model. I don't want to mix key for fetching and real model object.

File: app/assets/javascripts/components/events/Events.js.jsx (hunk 7 above)
Comment (irynar) [inline app/assets/javascripts/components/events/Events.js.jsx]: done :)

File: app/controllers/application_controller.rb (hunk 5 above)
Comment (irynar) [inline app/controllers/application_controller.rb]: ok

File: app/controllers/application_controller.rb (hunk 4 above)
Comment (irynar) [inline app/controllers/application_controller.rb]: ok

File: app/views/application/index.html.erb (hunk 3 above)
Comment (irynar) [inline app/views/application/index.html.erb]: ok

**Comment by irynar** [comment]
I added all changes. Please review and merge.

**Comment by irynar** [comment]
Rebased. Please review and merge.

File: app/assets/javascripts/components/events/FieldEdit.js.jsx [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+    return (
+      <div className="form-group">
+        <label>{label}:</label>
+        { formatted }
+      </div>
+    );
+  }
+});
+
+function capitalize(text) {
```
Comment (rtriska) [inline app/assets/javascripts/components/events/FieldEdit.js.jsx]: Move to helpers :+1:

**Comment by rtriska** [comment]
Code lgtm @forecome  please merge

**Pull request merged**